pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go

```diff
@@ -202,11 +200,36 @@ func (r *ReconcileNvmeOfStorage) Reconcile(context context.Context, request reco
 		}
 
 		return reporting.ReportReconcileResult(logger, r.recorder, request, r.nvmeOfStorage, reconcile.Result{}, err)
+	} else if strings.Contains(request.Name, "rook-ceph-osd") {
```

Comment:
How do we know that this event (sting = rook-ceph-osd) is originated from the osd fault?

Reply:
because there are only two events registered to call the reconciler: nvmeofstorage and rook-ceph-osd. 

This is evident from the code:
```
cmKind := source.Kind(
    mgr.GetCache(),
    &cephv1.NvmeOfStorage{TypeMeta: controllerTypeMeta})
err = c.Watch(cmKind, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())

podKind := source.Kind(
    mgr.GetCache(),
    &corev1.Pod{})
err = c.Watch(podKind, &handler.EnqueueRequestForObject{},
```
These lines of code show that only events related to nvmeofstorage and rook-ceph-osd are set up to trigger the reconciler.

Reply:
Got it. Hmm, Is there any better way to identify where the event is originated from? I think this string parser is not a good way to go. Al very least, please add a TODO.

Reply:
Okay. I added a TODO to refactor this part of the code to use a more reliable method